Bookmarking

AnnekeSchuette opened the pull request:
I added a bookmark function to my WineCard component including unit tests.
Thanks in advance for your feedback and improvement suggestions! 💣🚀🙏🏻

## Value statement

As a **user** 
I **need a bookmarking function on wine cards**
so that I can  ** add** them to a list of saved wines to quickly find them later on

## Acceptance criteria
- every WineCard has a Bookmark Icon 
- the bookmark-icon (heart) has two visual states: active / inactive (default)
- the state toggles on click
- bookmarked wines are **saved** into an array _savedWines_
- de-bookmarked Wines are **removed** from the array _savedWines_

VictoriaWika reviewed (commented):
Nice work! 👍🏻 I left some small comments

AnnekeSchuette commented on client/src/components/App/App.js:
```diff
@@ -3,13 +3,17 @@ import Header from 'components/Header/Header'
 
 export default function App() {
+  const [savedWines, { handleSaveWine }] = useSaveWine()
+  console.log(savedWines)
```
😅

AnnekeSchuette commented on client/src/components/WineCard/WineCard.js:
```diff
@@ -30,12 +35,19 @@ export default function WineCard({
+      <BookmarkButton
+        onClick={() => handleBookmarkClick({ id, title })}
+        isActive={toggleState}
+        aria-label="Bookmark"
```
Thanks a lot for this helpful advice! I added the aria-label for testing purposes, but didn't really consider the value for the user.

AnnekeSchuette commented:
> Nice work! 👍🏻 I left some small comments

@VictoriaWika Thanks a lot for your code review! I've made a few improvements based on your comments, fixed storybook and added a new test. It would be great, if you could approve the changes regarding your annotations. Merci! 🙏🏻

https://github.com/AnnekeSchuette/capstone-project/compare/6e708fbf75e3b2af5f6d6e8063f71c5c9f050686...0525124eff2af2873da312f25fd66fd5f106a791

VictoriaWika commented:
Great Job! Everything looks good! 👍🏻
